framework/signature-inference-scripts/infer-and-annotate.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/sh
+
+# This script runs the Checker Framework's signature inference
+# iteratively on a program, adding type annotations on the program, until the
+# .jaif files from one iteration are the same as the .jaif files from the
+# previous iteration.
+
+# This script receives as arguments:
+# 1. Processor's name.
+# 2. Classpath (target project's classpath).
+# 3. Extra processor arguments which will be passed to the checker.
+# 4. List of paths to .java files in a program.
+
+# Example of usage:
+# ./infer-and-annotate.sh "LockChecker,NullnessChecker" \
+#     $JSR308/plume-lib/java/plume.jar -AprintErrorStack \
+#     `find $JSR308/plume-lib/java/src/plume/ -name "*.java"`
+
+# In case of using this script for Android projects, you must include paths to:
+# android.jar, gen folder, all libs used, source code folder.
+# The Android project must be built with "ant debug" before running this script.
+
+# Halts the script when a nonzero value is returned from a command.
+set -e
+
+# This function separates extra arguments passed to the checker from java files
+# received as arguments.
+# TODO: Handle the following limitation: This function makes the assumption
+# that every argument starts with a hyphen. It means one cannot pass arguments
+# such as -processorpath and -source.
+read_input() {
+    processor=$1
+    cp=$2:${CHECKERFRAMEWORK}/../annotation-tools/annotation-file-utilities/annotation-file-utilities.jar
+    extra_args=""
+    java_files=""
+
+    # Ignores first two arguments (processor and cp).
```

Comment:
I would move these two lines (and the blank line below them) between lines 33 and 34, thus putting all the processing of the first two arguments together.